src/Abstractions/NexusMods.Abstractions.Settings/ServiceExtensions.cs

```diff
@@ -12,12 +12,26 @@ public static class ServiceExtensions
     /// <summary>
     /// Registers a settings type.
     /// </summary>
-    public static IServiceCollection AddSettings<T>(this IServiceCollection serviceCollection) where T : class, ISettings, new()
+    public static IServiceCollection AddSettings<T>(this IServiceCollection serviceCollection)
+        where T : class, ISettings, new()
     {
         return serviceCollection.AddSingleton(new SettingsTypeInformation(
             ObjectType: typeof(T),
             DefaultValue: new T(),
             ConfigureLambda: T.Configure
         ));
     }
+
+    /// <summary>
+    /// Registers an override for <typeparamref name="T"/>.
+    /// </summary>
+    public static IServiceCollection OverrideSettings<T>(
```

Comment:
I think the comment here needs to be a bit more forthcoming in what this is doing. Here or someplace else.

From my understanding it is creating a fake layer for the setting type that takes precedence over the actual saved settings, without actually changing the original settings in whatever datastore we are using.

This mechanism needs to be advertised or explained someplace I feel, and the SettingsManager is hiding away all of this in implementation